crates/fsrt/src/main.rs

```diff
@@ -99,7 +105,90 @@ impl fmt::Display for Error {
     }
 }
 
-impl std::error::Error for Error {}
+fn check_graphql_and_perms(val: &Value) -> Vec<&str> {
+    let mut operations = vec![];
+    match val {
+        Value::Const(Const::Literal(s)) => operations.extend_from_slice(&parse_graphql(s)),
+        Value::Phi(vals) => vals.iter().for_each(|val| match val {
+            Const::Literal(s) => operations.extend_from_slice(&parse_graphql(s)),
+        }),
+        _ => {}
+    }
+    // TODO : Build out permission resolver here
+
+    let permissions_resolver: HashMap<(String, String), &str> = [(
```

Comment:
I think the first entry `compass` of the tuple key could be an enum of platforms that we support, since this will probably be static even in the final approach.

Also, does the second entry need to be a `String`?

Reply:
oh no, this is your day off 😱. sorry

I think it does because when you are iterating through the `doc.definitions` on line 144 it requires you to move the the selection and fragment's ownership into the scope. Then, later you would need to return a reference to data in the fragment potentially, which, at that point, it would be owned by the inner scope. As a result, you need to declare the memory on the heap and pass it back. Unless, I am missing something.

I spent 2 hours trying to do that this morning 💀

Reply:
Ah, can't you just return `Vec<(&str, &str)>` directly from `parse_graphql`? The only reason why you need to convert it to a `String` is because of `get_field_and_operation`.